docs/blog-posts/symmetric-key-support.md

```diff
@@ -0,0 +1,220 @@
+        if (secret?.Value == null)
+            throw new InvalidOperationException($"Secret {description.KeyVaultSecretName} not found in Key Vault");
+
+        try
+        {
+            // Convert secret value to bytes and create SymmetricSecurityKey
+            var keyBytes = Convert.FromBase64String(secret.Value.Value);
+            description.CachedValue = new SymmetricSecurityKey(keyBytes);
+        }
+        catch (Exception ex)
+        {
+            throw new InvalidOperationException($"Failed to create symmetric key from Key Vault secret: {ex.Message}", ex);
+        }
+    }
+}
+
+internal class Base64EncodedSymmetricKeyLoader : ICredentialSourceLoader
+{
+    public async Task LoadIfNeededAsync(CredentialDescription description, CredentialSourceLoaderParameters? parameters)
+    {
+        _ = Throws.IfNull(description);
+
+        if (description.CachedValue != null)
+            return;
+
+        if (string.IsNullOrEmpty(description.Base64EncodedValue))
+            throw new ArgumentException("Base64EncodedValue is required for Base64Encoded source");
+
+        try
+        {
+            // Convert Base64 string to bytes and create SymmetricSecurityKey
+            var keyBytes = Convert.FromBase64String(description.Base64EncodedValue);
+            description.CachedValue = new SymmetricSecurityKey(keyBytes);
+        }
+        catch (Exception ex)
+        {
+            throw new FormatException("Invalid Base64 string for symmetric key", ex);
+        }
+
+        await Task.CompletedTask.ConfigureAwait(false);
+    }
+}
+```
+
+### 4. DefaultCredentialsLoader Changes(IdWeb Layer)
+Update the loader to handle both certificate and symmetric key scenarios:
+
+```csharp
+public partial class DefaultCredentialsLoader : ICredentialsLoader, ISigningCredentialsLoader
+{
+    public DefaultCredentialsLoader(ILogger<DefaultCredentialsLoader>? logger)
+    {
+        _logger = logger ?? new NullLogger<DefaultCredentialsLoader>();
+
+        CredentialSourceLoaders = new Dictionary<CredentialSource, ICredentialSourceLoader>
+        {
+            // Existing certificate loaders
+            { CredentialSource.KeyVault, new KeyVaultCertificateLoader() },
+            { CredentialSource.Path, new FromPathCertificateLoader() },
+            { CredentialSource.StoreWithThumbprint, new StoreWithThumbprintCertificateLoader() },
+            { CredentialSource.StoreWithDistinguishedName, new StoreWithDistinguishedNameCertificateLoader() },
+            { CredentialSource.Base64Encoded, new Base64EncodedCertificateLoader() },
+
+            // New symmetric key loaders
+            { CredentialSource.SymmetricKeyFromKeyVault, new KeyVaultSymmetricKeyLoader(_secretClient) },
+            { CredentialSource.SymmetricKeyBase64Encoded, new Base64EncodedSymmetricKeyLoader() }
+        };
+    }
+
+    public async Task<SigningCredentials?> LoadSigningCredentialsAsync(
```

Comment:
I wonder if a simpler design would be to rely on the existing object model, i.e. `ICredentialsLoader` and to treat the `SymmetricSecurityKey` as another type of credential - see `CredentialType` enum (alongside `DecryptKeys`, which is also not used by MSAL).

Apart from simpelr object model, this would avoid mixing the concept of `SigningCredentials` in Id.Web. I believe `SigningCredentials` + X509 has some perf implications (i.e. loading it often is expensive, caching needs to be done).